### PR TITLE
Add breadcrumbs documentation

### PR DESCRIPTION
**Pull request by alexandra-simeonova**
**Description**

Changes proposed in this pull request:

- Add breadcrumbs documentation

**Related issue(s)**
#2416

### CODE REVIEW

File: docs/navigation-advanced.md [hunk 1]
```diff
@@ -443,6 +444,104 @@ In the case the node has only one child, it's possible to configure if the horiz
...
+            omitRoot: false,
+            pendingItemLabel: '...',
+            renderer: (el, items, clickHandler) => {
+              el.classList.add('dxp-breadcrumb');
+              const ui5breadcrumbs =
+                el.querySelector('ui5-breadcrumbs') ||
+                document.createElement('ui5-breadcrumbs');
+              ui5breadcrumbs.innerHTML = '';
+              items.forEach((item, index) => {
+                const label: string = item.label;
```
Comment (ndricimrr) [inline docs/navigation-advanced.md:513]:
```suggestion
                const label = item.label;
```

File: docs/navigation-advanced.md [hunk 2]
```diff
@@ -443,6 +444,104 @@ In the case the node has only one child, it's possible to configure if the horiz
...
+                
+                if (label && !label.startsWith(':virtualSegment_')) {
+                  const itemCmp = document.createElement(
+                    'ui5-breadcrumbs-item',
+                  );
+                  itemCmp.setAttribute('href', item.route);
+                  itemCmp.innerHTML = label;
+                  (itemCmp as any)._item = item;
```
Comment (ndricimrr) [inline docs/navigation-advanced.md:521]:
```suggestion
                  itemCmp._item = item;
```

File: docs/navigation-advanced.md [hunk 3]
```diff
@@ -443,6 +444,104 @@ In the case the node has only one child, it's possible to configure if the horiz
...
+              const ui5breadcrumbs =
+                el.querySelector('ui5-breadcrumbs') ||
+                document.createElement('ui5-breadcrumbs');
+              ui5breadcrumbs.innerHTML = '';
+              items.forEach((item, index) => {
+                const label: string = item.label;
+                
+                if (label && !label.startsWith(':virtualSegment_')) {
+                  const itemCmp = document.createElement(
+                    'ui5-breadcrumbs-item',
```
Comment (ndricimrr) [inline docs/navigation-advanced.md:517]:
```suggestion
                    'ui5-breadcrumbs-item'
```

File: docs/navigation-advanced.md [hunk 4]
```diff
@@ -443,6 +444,104 @@ In the case the node has only one child, it's possible to configure if the horiz
...
+                  );
+                  itemCmp.setAttribute('href', item.route);
+                  itemCmp.innerHTML = label;
+                  (itemCmp as any)._item = item;
+                  ui5breadcrumbs.appendChild(itemCmp);
+                }
+              });
+              ui5breadcrumbs.addEventListener('item-click', (event: any) => {
```
Comment (ndricimrr) [inline docs/navigation-advanced.md:525]:
```suggestion
              ui5breadcrumbs.addEventListener('item-click', (event) => {
```

**Review by ndricimrr — COMMENTED** [review]
Some small comment to remove Typescript typings as this makes it harder to try out if copying it to Fiddle or a non-Typescript example

File: docs/navigation-advanced.md [hunk 5]
```diff
@@ -443,6 +444,104 @@ In the case the node has only one child, it's possible to configure if the horiz
   ...
 ```
 
+## Breadcrumbs
+
+
+
+Luigi allows you to add [breadcrumbs](https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Breadcrumb_Navigation) to your application. First, you need to create your own custom code implementing breadcrumbs, then enable them on the node level using Luigi's [showBreadcrumbs](navigation-parameters-reference.md#showbreadcrumbs) parameter. In your custom code, you can choose any look and style for the breadcrumbs as well as define what should happen upon clicking them. However, the code should follow this general pattern and return the variable `breadcrumbs`:
```
Comment (ndricimrr) [inline docs/navigation-advanced.md:451]:
```suggestion
Luigi allows you to add [breadcrumbs](https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Breadcrumb_Navigation) to your application. You need to create your own custom code implementing breadcrumbs. Once the breadcrumbs config is set it is enabled by default for all nodes. If you wish to disable it for a particular node you need to set [showBreadcrumbs](navigation-parameters-reference.md#showbreadcrumbs): true for that node.
The breadcrumbs configuration structure looks like this:

      navigation.breadcrumbs = {
            omitRoot: false,  // if set to true, the root node in breadcrumb hierarchy is omitted 
            clearBeforeRender: true, // if set to true, the containerElement will be cleared first, before being rendered. If set to false, handling of the clear before render needs to be handled by your side 
            autoHide: true, // hide breadcrumbs when navigating to root node
            renderer: (containerElement, nodeItems, clickHandler)  => {
                  // containerElement - refers to HTML element that contains the breadcrumb structure to which you can append your own customised elements
                  // nodeItems : {
                       label: label of node,
                       node: navigation node,
                       route: node route,
                       pending: indicates whether node label has not been resolved yet or not
                  }
                  // clickHandler(node) , can be called per node, to navigate to that node
            }
      }


. In your custom code, you can choose any look and style for the breadcrumbs as well as define what should happen upon clicking them. However, the code should follow this general pattern and return the variable `breadcrumbs`: 
```

File: docs/navigation-advanced.md [hunk 6]
```diff
@@ -443,6 +444,104 @@ In the case the node has only one child, it's possible to configure if the horiz
...
+}
+```
+
+<!-- accordion:start -->
+Below is an example of a simple `breadcrumbsConfig`:
+
+### Click to expand
+
+```js
+var breadcrumbsConfig = {
```
Comment (ndricimrr) [inline docs/navigation-advanced.md:470]:
```suggestion
navigation.breadcrumbs = {
```

File: docs/navigation-parameters-reference.md [hunk 7]
```diff
@@ -605,6 +605,18 @@ runTimeErrorHandler: {
   - A static **pathSegment** of value `settings` results in `example.com/settings`.
   - A dynamic **pathSegment** is prefixed with a colon and can load any value. Find out more about dynamic paths in Luigi [here](navigation-advanced.md#dynamically-changeable-paths).
 
+### showBreadcrumbs
+- - **type**: boolean
+- **description**: shows a breadcrumbs header for the node. See the [advanced navigation](navigation-advanced.md#breadcrumbs) document for more information.
```
Comment (ndricimrr) [inline docs/navigation-parameters-reference.md:610]: if breadcrumbs configuration is setup, the breadcrumbs show for all nodes. This property allows disabling the breadcrumb for one particular node by setting it to `false` for that node.
```suggestion
- **description**: if breadcrumbs configuration is setup, the breadcrumbs show for all nodes. This property allows disabling the breadcrumb for one particular node by setting it to `false` for that node.. See the [advanced navigation](navigation-advanced.md#breadcrumbs) document for more information.
```

File: docs/navigation-parameters-reference.md [hunk 8]
```diff
@@ -605,6 +605,18 @@ runTimeErrorHandler: {
...
+- - **example**:
+```javascript
+navigation: {
+  nodes: [{
+    pathSegment: 'home',
+    showBreadcrumbs: true,
```
Comment (ndricimrr) [inline docs/navigation-parameters-reference.md:616]:
```suggestion
    showBreadcrumbs: false, 
```

**Review by ndricimrr — APPROVED** [review]
Looks good 👍

**Comment by JohannesDoberer** [comment]
Blocked by #3198

File: docs/navigation-advanced.md [hunk 9]
```diff
@@ -443,6 +444,114 @@ In the case the node has only one child, it's possible to configure if the horiz
...
+          event.preventDefault();
+          clickHandler(event.detail.item._item);
+        });
+        el.appendChild(breadcrumbs);
+        return breadcrumbs;
+      }
+    };
+```
+
+Below is another example which uses UI5 breadcrumbs:
```
Comment (hardl) [inline docs/navigation-advanced.md:507]: ... uses UI5 Web Components breadcrumbs:

**Review by hardl — CHANGES_REQUESTED** [review]

File: docs/navigation-advanced.md (hunk 9 above)
Comment (alexandra-simeonova) [inline docs/navigation-advanced.md:507]:
```suggestion
Below is another example which uses UI5 Web Components breadcrumbs: 
```

**Review by hardl — APPROVED** [review]

**Pull request merged**